### PR TITLE
feature: UI Redesign Using Tailwind + Accessibility enhancements

### PR DESCRIPTION
**Pull request by jorgemxm**
@diego0020 Please Review:

wave-01

wave-02

### CODE REVIEW

File: wave-fe/package.json [hunk 1]
```diff
@@ -20,17 +20,19 @@
...
     "@angular/router": "~9.1.9",
     "firebase": "^7.15.0",
+    "focus-visible": "^5.1.0",
     "rxjs": "~6.5.4",
     "tslib": "^1.10.0",
     "zone.js": "~0.10.2"
   },
   "devDependencies": {
+    "@angular-builders/custom-webpack": "^9.2.0",
     "@angular-devkit/build-angular": "~0.901.7",
```
Comment (jorgemxm) [inline wave-fe/package.json:30]: Perhaps we could remove this `"@angular-devkit/build-angular"` module, pending to Test/Review it.

File: wave-fe/src/app/app.component.ts [hunk 2]
```diff
@@ -1,18 +1,15 @@
 import { Component, OnInit } from '@angular/core';
-import * as firebase from 'firebase/app';
 import 'firebase/database';
 import 'firebase/auth';
-import { RoundService } from './round.service';
```
Comment (jorgemxm) [inline wave-fe/src/app/app.component.ts:5]: These modules where not being used anywhere within this file. (ESLint Warning)

File: wave-fe/src/app/clues/clues.component.ts [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+  constructor(private guessService: GuessService, private roundService: RoundService) {
+    this.isGuestPlayer$ = roundService.isGuestPlayer$;
+  }
+
+  get startClasses() {
+    const color = this.clues?.startColor;
+    return `bg-${color}-400 text-${color}-400`;
+  }
+
+  get endClasses() {
```
Comment (jorgemxm) [inline wave-fe/src/app/clues/clues.component.ts:23]: Should it be `get endClasses(): string {` @diego0020 ?

File: wave-fe/src/scss/tailwind/custom-components.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  @apply px-4;
+  @apply py-2;
+  @apply rounded;
+  @apply text-black;
+  @apply text-blue-300;
+  transition: 200ms;
+}
+
+.btn:hover {
+  @apply bg-gray-500;
```
Comment (jorgemxm) [inline wave-fe/src/scss/tailwind/custom-components.scss:13]: @bg-gray-500 line can be removed, its being overwritten by the line below

File: wave-fe/src/scss/tailwind/custom-components.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
+.btn {
+  @apply bg-gray-400;
```
Comment (jorgemxm) [inline wave-fe/src/scss/tailwind/custom-components.scss:2]: @bg-gray-400 line can be removed, its being overwritten by the line below;

File: wave-fe/src/scss/tailwind/custom-components.scss [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  @apply bg-blue-800;
+  @apply px-4;
+  @apply py-2;
+  @apply rounded;
+  @apply text-black;
```
Comment (jorgemxm) [inline wave-fe/src/scss/tailwind/custom-components.scss:7]: `@apply text-black` line can be removed, its being overwritten by the line below

File: wave-fe/src/styles.scss [hunk 7]
```diff
@@ -1,105 +1,316 @@
...
 
 .board {
+  display: flex;
+  flex: 60% 1;
+  position: relative;
+  border-bottom: 1px solid $game-bg-dark;
+
+  &::before {
+    content: "";
+    // background-color: darken($game-bg, 3%);
```
Comment (jorgemxm) [inline wave-fe/src/styles.scss:67]: Remove unused code

File: wave-fe/src/styles.scss [hunk 8]
```diff
@@ -1,105 +1,316 @@
...
+
+  &__buttons-increase {
+    right: theme('spacing.12');
+  }
+
+  &__button {
+    background-color: darken($game-bg, 5%);
+    @apply rounded mx-1 p-2;
+    // @apply mx-1;
+    // @apply p-2;
```
Comment (jorgemxm) [inline wave-fe/src/styles.scss:196]: Remove Unused code

File: wave-fe/src/styles.scss [hunk 9]
```diff
@@ -1,105 +1,316 @@
...
+
+    background-color: $target-card-bg;
+    border-color: darken($target-card-bg, 5%);
+    color: lighten($game-bg, 40%);
+    @apply border flex mx-2 px-2 py-2 rounded;
+    // @apply flex;
+    // @apply mx-2;
+    // @apply px-2;
+    // @apply py-2;
+    // @apply rounded;
```
Comment (jorgemxm) [inline wave-fe/src/styles.scss:301]: Remove unused code

File: wave-fe/src/app/round.service.ts [hunk 10]
```diff
@@ -16,29 +18,64 @@ export class RoundService {
   private privateAddr = `roundsPrivate/${this.roundId}`;
   private roundRef = firebase.database().ref(this.roundAddr);
   private roundSubject = new ReplaySubject<any>(1);
-  private newRoundData: any = null;
   private score: any = null;
+  private newRoundData: {
```
Comment (jorgemxm) [inline wave-fe/src/app/round.service.ts:22]: Added for documentation purposes, Its easier to understand what the initial state is.

File: wave-fe/src/app/round.service.ts [hunk 11]
```diff
@@ -16,29 +18,64 @@ export class RoundService {
...
       const roundData = snapshot.val();
-      console.log(roundData, 'roundd ata');
+      console.log(roundData, 'round data');
       this.roundSubject.next(
         this.procRoundData(roundData)
       );
     });
   }
 
+  private generateRandData() {
```
Comment (jorgemxm) [inline wave-fe/src/app/round.service.ts:55]: Should it be `private generateRandData(): Object`? or should we create an Interface for this Object? @diego0020

File: wave-fe/src/app/clues/clues.component.ts (hunk 3 above)
Comment (diego0020) [reply wave-fe/src/app/clues/clues.component.ts:23]: yes

File: wave-fe/src/app/round.service.ts (hunk 11 above)
Comment (diego0020) [reply wave-fe/src/app/round.service.ts:55]: interface would be the best

File: wave-fe/src/app/game-info/game-info.component.ts [hunk 12]
```diff
@@ -14,8 +14,8 @@ export class GameInfoComponent implements OnInit {
   userId$: Observable<any>;
   round$: Observable<any>;
 
-  constructor(private round: RoundService, auth: AuthService) {
-    this.round$ = this.round.round$;
+  constructor(private roundService: RoundService, auth: AuthService) {
+    this.round$ = this.roundService.round$;
```
Comment (diego0020) [inline wave-fe/src/app/game-info/game-info.component.ts:18]: I don't know, maybe better to let consumer who need access to round observable to inject roundService

File: wave-fe/package.json (hunk 1 above)
Comment (diego0020) [reply wave-fe/package.json:30]: yes, this is an important test

File: wave-fe/src/scss/tailwind/custom-components.scss (hunk 4 above)
Comment (jorgemxm) [reply wave-fe/src/scss/tailwind/custom-components.scss]: Fixed

File: wave-fe/src/scss/tailwind/custom-components.scss (hunk 5 above)
Comment (jorgemxm) [reply wave-fe/src/scss/tailwind/custom-components.scss]: Fixed

File: wave-fe/src/scss/tailwind/custom-components.scss (hunk 6 above)
Comment (jorgemxm) [reply wave-fe/src/scss/tailwind/custom-components.scss]: fixed

File: wave-fe/src/styles.scss (hunk 8 above)
Comment (jorgemxm) [reply wave-fe/src/styles.scss]: fixed

File: wave-fe/src/styles.scss (hunk 7 above)
Comment (jorgemxm) [reply wave-fe/src/styles.scss]: Fixed

File: wave-fe/src/styles.scss (hunk 9 above)
Comment (jorgemxm) [reply wave-fe/src/styles.scss]: Fixed

File: wave-fe/src/app/round.service.ts (hunk 11 above)
Comment (jorgemxm) [reply wave-fe/src/app/round.service.ts]: I've Added a interface, please review latest version :)

File: wave-fe/package.json (hunk 1 above)
Comment (jorgemxm) [reply wave-fe/package.json:30]: Could you please test it if you have some spare time ? 🤔

File: wave-fe/src/app/clues/clues.component.ts (hunk 3 above)
Comment (jorgemxm) [reply wave-fe/src/app/clues/clues.component.ts]: Updated

File: wave-fe/src/app/game-info/game-info.component.ts (hunk 12 above)
Comment (jorgemxm) [reply wave-fe/src/app/game-info/game-info.component.ts:18]: I don't understand your comment :|

Comment (diego0020) [reply wave-fe/src/app/game-info/game-info.component.ts:18]: wrong place

Comment (diego0020) [reply wave-fe/src/app/game-info/game-info.component.ts:18]: never mind, it is correct, I was confusing services

**Pull request merged**